ILL/IDF/common_IDF_functions.py

```diff
@@ -0,0 +1,191 @@
+"""This Python script contains a number of utility functions for
+   creating IDF files for the ILL.
+"""
+
+from __future__ import print_function
+from time import gmtime, strftime
+import math
+import numpy
+from scipy.constants import degree
+from shutil import copyfile
+import os
+
+
+def read_detector_box_list(filename):
+    """Reads a text file containing rows with box Number, Theta, Detector IDs and Phi angle. The first
+       line is ignored (column headings).
+
+       returns: A list of tuples containing box_number, theta angle, a list of detector ids, phi angle
+    """
+
+    box_number = []
+    theta = []
+    detector_ids = []
+    phi = []
+
+    with open(filename) as detector_box_list:
+        next(detector_box_list)  # Skip header line
+
+        for line in detector_box_list:
+            line_entries = line.split()
+
+            box_number.append(int(line_entries[0]))
+            theta.append(float(line_entries[1]))
+
+            detector_id_line = []
+            for value in line_entries[2].split(','):
+                try:
+                    detector_id_line.append(int(value))
+                except:
+                    pass
+            detector_ids.append(detector_id_line)
+            phi.append(float(line_entries[3]))
+
+    return zip(box_number, theta, detector_ids, phi)
+
+
+def write_header(output_file, instrument_name, authors, valid_from='1900-01-31 23:59:59'):
+    output_file.write('<?xml version="1.0" encoding="UTF-8"?>\n')
```

Comment:
Can I suggest using the python `lxml` module here? For example see https://github.com/mantidproject/mantidgeometry/blob/master/helper.py#L1. It's generally considered better practice to use these types of library for serializing and deserializing content to and from strings.